### PR TITLE
Pass the memory handler and filter config to exclude cuda transport

### PR DESCRIPTION
**Pull request by tvegas1**
Due to parallel cuda kernel running, UCX plugin must not try to use cuda primitives to manipulate passed cuda memory. To do so we:
1. Disable `cuda` transport alias altogether
2. Don't try to resolve the cuda memory chunk to the whole cuda memory area
3. Force RNDV to avoid the need for explicit memory copy (eager, rx path)
4. Pass memory handle to allow memory details to be looked-up instead

### CODE REVIEW

**Comment by tvegas1** [comment]
pls look @brminich, @yosefe

File: src/ucx_plugin.c [hunk 1]
```diff
@@ -210,14 +210,43 @@ static ncclResult_t GetSocketAddr(union ncclSocketAddress *addr) {
...
+  const char *ucx_tls;
+  ssize_t n;
+
+  ucx_tls = getenv("NCCL_UCX_TLS");
+  if (ucx_tls == NULL) {
+    ucx_tls = getenv("UCX_TLS");
+  }
+
+  if (ucx_tls != NULL) {
+    n = snprintf(tls, sizeof(tls), "^cuda,%s", ucx_tls);
```
Comment (brminich) [inline src/ucx_plugin.c:224]: afair this will discard all transports listed, including user's config

Comment (tvegas1) [reply src/ucx_plugin.c]: now checking if specification is positive, negative or default

**Comment by tvegas1** [comment]
> lgtm, but can we make it configurable just in case? I.e. have a single NCCL var disabling or enabling cuda support in UCX plugin

as discussed, added complete bypass config possibility to prevent blocking any further testing down the line.

**Review by brminich — APPROVED** [review]

File: src/ucx_plugin.c [hunk 2]
```diff
@@ -212,15 +214,58 @@ static void recv_handler_nbx(void *request, ucs_status_t status,
...
+  } else {
+    /* Positive expression cannot allow cuda-like transports */
+    if ((strstr(ucx_tls, "cuda") != NULL) || (strstr(ucx_tls, "gdr") != NULL)) {
+      WARN("Cannot use cuda/gdr transports as part of specified UCX_TLS");
+      return ncclInternalError;
+    }
+  }
+
+  UCXCHECK(ucp_config_modify(config, "TLS", ucx_tls));
+  UCXCHECK(ucp_config_modify(config, "RNDV_THRESH", "0"));
```
Comment (brminich) [inline src/ucx_plugin.c:246]: btw, i think we also have to disable am-data scheme in RNDV protocols, i.e. set `UCX_RNDV_SCHEME=get_zcopy,put_zcopy`

**Review by brminich — CHANGES_REQUESTED** [review]

File: src/ucx_plugin.c (hunk 2 above)
Comment (tvegas1) [reply src/ucx_plugin.c:246]: added "RNDV_SCHEME=get_zcopy" as the configuration does not take a list.

Comment (tvegas1) [reply src/ucx_plugin.c:246]: fixed with `get_zcopy` as only one parameter can be specified

**Review by brminich — APPROVED** [review]

**Review by bureddy — APPROVED** [review]

**Pull request merged**